packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -87,14 +100,25 @@ export class FileMenuContribution implements MenuContribution {
 @injectable()
 export class WorkspaceCommandContribution implements CommandContribution {
 
+    protected roots: FileStat[] = [];
```

Comment:
Would it be possible to get rid of unnecessary state and read it each time from `WorkspaceService`? If not maybe we can cache them somewhere, like to have `WorkspaceService.tryWorksace` which returns latest roots, meaning an empty list if none were yet resolved

Reply:
good idea. I added `tryGetWorksace()` to `WorkspaceService`